### ISSUE TITLE
An addon to customize the areas in the editor

### ISSUE THREAD

**Issue by fractioneater** [ScratchAddons/ScratchAddons#5544]
### Why this would be helpful

The space on the right of the editor seems limited. It always has the stage and sprite pane, and there's not really anything you can do to change that, except resizing it (and dragging the debugger on top of it, I guess). I sometimes have a need for the stage, debugger, and code area, but not the sprite pane. I also sometimes have a need for the sprite pane, variable manager, and code area, but not the stage. This addon would be hard to implement, but it would give users full control of what they can see.

### How this addon works

A bar is added to the right of the editor that can be used to collapse areas and change what is occupying the space.

For example, let's say I wanted to see the local variables for multiple sprites. I would switch out the stage for the variable manager, and now I can see the code area, sprite pane, and variable manager.

After that, I might want to have more space to code, but be able to switch sprites. I would collapse the variable manager, and resize the sprite pane so it doesn't take up as much space.

Third, (don't worry, this is the last example) if I wanted to use the console temporarily, I would replace the upper and lower areas with the console, and it would automatically fill all of the vertical space.

### Possible alternatives

This could be part of the debugger addon, but it's only similar in a few ways.

### Additional context

This was inspired by two things: Chrome devtools, and a Scratch mod that has a collapsible project player and sprite pane. [Here's the link](https://cocrea.world/gandi).

**Comment by RIMOPA** [comment]
Realted: #5464

**Comment by fractioneater** [comment]
> Related: #5464

Not really, the fullscreen editor only modifies the stage and sprite pane together, and they aren't changed in the way I was describing.

**Comment by RIMOPA** [comment]
> > Related: #5464
> 
> Not really, the fullscreen editor only modifies the stage and sprite pane together, and they aren't changed in the way I was describing.

But we will not do two addons that can do the same, so I was referring to what we would do with the other addon. I think we would delete it or just add the fullscreen button.

**Comment by fractioneater** [comment]
> But we will not do two addons that can do the same, so I was referring to what we would do with the other addon. I think we would delete it or just add the fullscreen button.

This doesn't do the same thing.

Also, there's already the hide stage and sprite pane addon, which actually does just do the same as fullscreen editor.

**Comment by RIMOPA** [comment]
I mean this addon could do that

> > But we will not do two addons that can do the same, so I was referring to what we would do with the other addon. I think we would delete it or just add the fullscreen button.
> 
> This doesn't do the same thing.
> 
> Also, there's already the hide stage and sprite pane addon, which actually does just do the same as fullscreen editor.

I mean this addon could. I wouldn't like having more than one addon that could do the same, so I would like this or the other addons to be deleted, merged or something, for example, i think merging this one with the hide stage and sprite pane one with a hide stage and hide sprite pane for default would be ideal.
